src/metaswitch/clearwater/config_manager/test/test_config_type.py

```diff
@@ -0,0 +1,182 @@
+        mock_xml.return_value = ['script3', 'script4'], ['name3', 'name4']
+        mock_shared.return_value = ['script5', 'script6'], ['name5', 'name6']
+
+        bgcf_json = bgcf_json_config_plugin.BgcfJson('path')
+        ifcs_xml = shared_ifcs_config_plugin.SharedIfcsXml('path')
+        weatherwax = weatherwax_config_plugin.WeatherwaxConfig('path')
+        shared_config = shared_config_config_plugin.SharedConfig('path')
+
+        bgcf_fake_scripts = ['script1', 'script2']
+        ifcs_fake_scripts = ['script3', 'script4']
+        weatherwax_fake_scripts = []
+        shared_fake_scripts = ['script5', 'script6']
+
+        bgcf_fake_names = ['name1', 'name2']
+        ifcs_fake_names = ['name3', 'name4']
+        weatherwax_fake_names = []
+        shared_fake_names = ['name5', 'name6']
+
+        self.assertListEqual(bgcf_json.scripts, bgcf_fake_scripts)
+        self.assertListEqual(ifcs_xml.scripts, ifcs_fake_scripts)
+        self.assertListEqual(weatherwax.scripts, weatherwax_fake_scripts)
+        self.assertListEqual(shared_config.scripts, shared_fake_scripts)
+
+        self.assertListEqual(bgcf_json.script_names, bgcf_fake_names)
+        self.assertListEqual(ifcs_xml.script_names, ifcs_fake_names)
+        self.assertListEqual(weatherwax.script_names, weatherwax_fake_names)
+        self.assertListEqual(shared_config.script_names, shared_fake_names)
+
+
+class TestGetValidationScripts(unittest.TestCase):
+    def test_get_json(self):
+        """tests the get_json_validation returns the correct list of scripts
+        where each script is a list"""
+        # currently only one
+        scscf_json = scscf_json_config_plugin.ScscfJson('path')
+        answer = scscf_json.get_json_validation()[0]
+        scscsf_expected_script = [['python',
+                                   '/usr/share/clearwater/clearwater-config-manager/scripts/validate_json.py',
+                                   '/usr/share/clearwater/clearwater-config-manager/scripts/config_validation/scscf_schema.json',
+                                   'path']]
+        self.assertListEqual(answer, scscsf_expected_script)
+
+    def test_get_xml(self):
+        """tests the get_xml_validation returns the correct list of scripts
+        where each script is a list"""
+        # currently only one
+        fallback_ifcs = fallback_ifcs_config_plugin.FallbackIfcsXml('path')
+        answer = fallback_ifcs.get_xml_validation()[0]
+        ifcs_expected_script = [['xmllint', '--format', '--pretty', '1',
+                                 '--load-trace', '--debug', '--schema',
+                                 '$/usr/share/clearwater/clearwater-config-manager/scripts/config_validation/fallback_ifcs_schema.xsd',
+                                 '$path', ]]
+        self.assertListEqual(answer, ifcs_expected_script)
+
+    @mock.patch("metaswitch.clearwater.config_manager.config_type_class_plugin.os")
+    def test_get_shared(self, mock_os):
```

Comment:
You aren't really testing that `get_sharedconfig_validation` function works the way you want it to here - all you're doing is checking that it looks in a directory at some point.

I would just scrap this and cover it as part of the `TestSharedValidation` tests below.